plugin v1.4.14 crushes

When trying to apply the following TF:
main.tf
```
locals {
  env0_api_endpoint            = format("https://api-%s.dev.env0.com", var.env0_agent_stage)
  docker_agent_k8s_secret_name = "docker-agent-bors-docker-configuration"
}

data "env0_agent_values" "agent_details" {
  agent_key = var.env0_agent_name
}

resource "kubernetes_secret" "docker_agent_bors_docker_configuration" {
  metadata {
    name = local.docker_agent_k8s_secret_name
  }
  type = "Opaque"
  data = {
    ENV0_STAGE                              = regex("(\"stage\": \")(.*)\"", data.env0_agent_values.agent_details.values)[1]
    ENV0_AGENT_KEY                          = regex("(\"agentKey\": \")(.*)\"", data.env0_agent_values.agent_details.values)[1]
    ENV0_ENCODED_DEPLOYMENT_TRIGGER_API_KEY = sensitive(regex("(\"env0ApiGwKeyEncoded\": \")(.*)\"", data.env0_agent_values.agent_details.values)[1])
    ENV0_ENCODED_AWS_ACCESS_KEY_ID          = regex("(\"awsAccessKeyIdEncoded\": \")(.*)\"", data.env0_agent_values.agent_details.values)[1]
    ENV0_ENCODED_AWS_SECRET_ACCESS_KEY      = sensitive(regex("(\"awsSecretAccessKeyEncoded\": \")(.*)\"", data.env0_agent_values.agent_details.values)[1])
  }
}

resource "kubernetes_pod" "docker-agent-bors" {
  metadata {
    name = var.env0_docker_agent_container_name
  }
  spec {
    container {
      image             = format("ghcr.io/env0/deployment-agent:%s", var.env0_agent_stage)
      name              = var.env0_docker_agent_container_name
      image_pull_policy = "Always"
      env {
        name  = "ENV0_STATE_ENCRYPTION_KEY"
        value = "cHIxMjc3Nw=="
      }
      env_from {
        secret_ref {
          name = local.docker_agent_k8s_secret_name
        }
      }
    }
  }
}
```

providers.tf
```
provider "aws" {
  region = var.aws_region
}

provider "env0" {
  api_key      = var.env0_agent_org_api_key
  api_secret   = var.env0_agent_org_api_secret
  api_endpoint = local.env0_api_endpoint
}

data "aws_eks_cluster" "cluster" {
  name = var.cluster_name
}

data "aws_eks_cluster_auth" "cluster" {
  name = var.cluster_name
}


provider "kubernetes" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  exec {
    api_version = "client.authentication.k8s.io/v1alpha1"
    args        = ["eks", "get-token", "--cluster-name", data.aws_eks_cluster.cluster.name]
    command     = "aws"
  }
}
```

versions.tf

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.68"
    }
    kubernetes = "~> 2.7.0"
    kubectl = {
      source  = "gavinbunney/kubectl"
      version = ">= 1.7.0"
    }
    env0 = {
      source = "env0/env0"
      version = ">= 1.4.14"
    }
  }
}
```

I get the following error:
```
╷
│ Error: Plugin did not respond
│ 
│   with provider["registry.terraform.io/env0/env0"],
│   on providers.tf line 5, in provider "env0":
│    5: provider "env0" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-env0_v1.4.14 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb238f8]

goroutine 7 [running]:
github.com/env0/terraform-provider-env0/env0.configureProvider.func1.1(0x0, {0x9a2775e108, 0x7f9a27767148})
	github.com/env0/terraform-provider-env0/env0/provider.go:165 +0x278
github.com/go-resty/resty/v2.Backoff(0xc0000f4f20, {0xc0000f4ef8, 0x5, 0xc0000126b0})
	github.com/go-resty/resty/v2@v2.7.0/retry.go:118 +0x40b
github.com/go-resty/resty/v2.(*Request).Execute(0xc0005b0820, {0xce3bf8, 0xc0000f4fc8}, {0xcec4c5, 0xe})
	github.com/go-resty/resty/v2@v2.7.0/request.go:763 +0x537
github.com/go-resty/resty/v2.(*Request).Get(...)
	github.com/go-resty/resty/v2@v2.7.0/request.go:689
github.com/env0/terraform-provider-env0/client/http.(*HttpClient).Get(0xc000218ec0, {0xcec4c5, 0xe}, 0xc0000f51c0, {0xb9e2a0, 0xc000011050})
	github.com/env0/terraform-provider-env0/client/http/client.go:95 +0x67f
github.com/env0/terraform-provider-env0/client.(*ApiClient).Organization(0xc0002c91a0)
	github.com/env0/terraform-provider-env0/client/organization.go:37 +0x9b
github.com/env0/terraform-provider-env0/client.(*ApiClient).OrganizationId(0xc0002c91a0)
	github.com/env0/terraform-provider-env0/client/organization.go:61 +0x78
github.com/env0/terraform-provider-env0/env0.configureProvider.func1({0xce59ce, 0x6}, 0xc0002c8d20)
	github.com/env0/terraform-provider-env0/env0/provider.go:198 +0x42a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc0000a0300, {0xe405c0, 0xc0002c8d20}, 0xd)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.22.0/helper/schema/provider.go:297 +0x1f5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0000103c0, {0xe405c0, 0xc0002c88a0}, 0xc000010f30)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.22.0/helper/schema/grpc_provider.go:557 +0x345
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc00025afa0, {0xe405c0, 0xc0002c81b0}, 0xc000218100)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:556 +0x2d3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0xcb3aa0, 0xc00025afa0}, {0xe405c0, 0xc0002c81b0}, 0xc0002d6000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004b0000, {0xe44000, 0xc0004b3040}, 0xc000358480, 0xc000221d40, 0x1322f58, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0004b0000, {0xe44000, 0xc0004b3040}, 0xc000358480, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-env0_v1.4.14 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/env0/provider.go
+++ b/env0/provider.go
@@ -162,7 +162,9 @@ func configureProvider(version string, p *schema.Provider) schema.ConfigureConte
 			SetRetryWaitTime(time.Second).
 			SetRetryMaxWaitTime(time.Second * 5).
 			OnBeforeRequest(func(c *resty.Client, r *resty.Request) error {
-				tflog.SubsystemInfo(subCtx, "env0_api_client", "Sending request", map[string]interface{}{"method": r.Method, "url": r.URL})
+				if r != nil {
+					tflog.SubsystemInfo(subCtx, "env0_api_client", "Sending request", map[string]interface{}{"method": r.Method, "url": r.URL})
+				}
 				return nil
 			}).
 			OnAfterResponse(func(c *resty.Client, r *resty.Response) error {
@@ -173,7 +175,7 @@ func configureProvider(version string, p *schema.Provider) schema.ConfigureConte
 			AddRetryCondition(func(r *resty.Response, err error) bool {
 				if r == nil {
 					// No response. Possiblly a networking issue (E.g. DNS lookup failure).
-					tflog.SubsystemWarn(subCtx, "env0_api_client", "No response, retrying request", map[string]interface{}{"method": r.Request.Method, "url": r.Request.URL})
+					tflog.SubsystemWarn(subCtx, "env0_api_client", "No response, retrying request")
 					return true
 				}
 

